lua/nvim-treesitter/refactor/navigation.lua

```diff
@@ -0,0 +1,87 @@
+-- Definition based navigation module
+
+local ts_utils = require'nvim-treesitter.ts_utils'
+local locals = require'nvim-treesitter.locals'
+local configs = require'nvim-treesitter.configs'
+local api = vim.api
+
+local M = {}
+
+local function node_to_qf(node, kind)
+  local lnum, col, _ = def.node:start()
+
+  return {
+    bufnr = bufnr,
+    lnum = lnum + 1,
+    col = col + 1,
+    text = ts_utils.get_node_text(def.node)[1] or '',
+    kind = kind
+  }
+end
+
+function M.goto_definition(bufnr)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  local node_at_point = ts_utils.get_node_at_cursor()
+
+  if not node_at_point then return end
+
+  local definition, _ = ts_utils.find_definition(node_at_point, bufnr)
+
+  if not definition then
+    print('No definition found')
```

Comment:
Should we output this as a warning ?

Reply:
What is the function for outputing a warning?

Reply:
i'm not sure there is one, you might need to output HL groups in the command line

Reply:
I changed it to `vim.fn.echoerr('No definition found')` Seems to get the job done...

Reply:
i got an unknown function for echoerr